### PR TITLE
Create applicationTrigger.py

### PR DESCRIPTION
**Pull request by istairbn**
Underlying code, no changes to the Synthetic. 

Allows an XL Release Template to be triggered if a new Application Version is added to XL Deploy. 
Useful for triggering processes where the pipeline does not start in XL Release (for example, starts from a CI build agent).

### CODE REVIEW

File: src/main/resources/xlr_xldeploy/applicationTrigger.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+
+from xlr_xldeploy.XLDeployClientUtil import XLDeployClientUtil
+
+xld_client = XLDeployClientUtil.create_xldeploy_client(xldeployServer, username, password)
+
+try:
+    response = xld_client.check_ci_exist(applicationId)
+except:
```
Comment (jdewinne) [inline src/main/resources/xlr_xldeploy/applicationTrigger.py]: check_ci_exist has a throw_on_fail option

**Comment by jdewinne** [comment]
Shouldn't this also include some changes in the `synthetic.xml`

File: src/main/resources/xlr_xldeploy/applicationTrigger.py (hunk 1 above)
Comment (istairbn) [reply src/main/resources/xlr_xldeploy/applicationTrigger.py]: Valid point - setting that to True. That means we can cut some of the fat off.

**Comment by istairbn** [comment]
Adding them in now

File: src/main/resources/xlr_xldeploy/applicationTrigger.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+from xlr_xldeploy.XLDeployClientUtil import XLDeployClientUtil
+
+xld_client = XLDeployClientUtil.create_xldeploy_client(xldeployServer, username, password)
+
+response = xld_client.check_ci_exist(applicationId,True)
+
+if not response:
+    raise Exception("%s does not exist" % (applicationId))
+
+packageId = xld_client.get_latest_package_version(applicationId)
```
Comment (jdewinne) [inline src/main/resources/xlr_xldeploy/applicationTrigger.py]: Can we refactor `packageId` into `package_id`?

**Review by jdewinne — CHANGES_REQUESTED** [review]
I think if you can make that last change, we're ready to merge this into master

**Pull request closed without merging**